src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1064 @@
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_REDUN_FAC") > 0) {	
+		type = DAOS_PROP_CO_REDUN_FAC;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_REDUN_FAC");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_REDUN_LVL") > 0) {	
+		type = DAOS_PROP_CO_REDUN_LVL;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_REDUN_LVL");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_SNAPSHOT_MAX") > 0) {	
+		type = DAOS_PROP_CO_SNAPSHOT_MAX;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_SNAPSHOT_MAX");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_COMPRESS") > 0) {	
+		type = DAOS_PROP_CO_COMPRESS;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_COMPRESS");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_ENCRYPT") > 0) {	
+		type = DAOS_PROP_CO_ENCRYPT;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_ENCRYPT");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_OWNER") > 0) {	
+		type = DAOS_PROP_CO_OWNER;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_str(file_id, entry, "DAOS_PROP_CO_OWNER");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_OWNER_GROUP") > 0) {	
+		type = DAOS_PROP_CO_OWNER_GROUP;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_str(file_id, entry,
+	    			     "DAOS_PROP_CO_OWNER_GROUP");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_DEDUP") > 0) {	
+		type = DAOS_PROP_CO_DEDUP;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_DEDUP");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_DEDUP_THRESHOLD") > 0) {	
+		type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_DEDUP_THRESHOLD");
+      		if (rc != 0) {
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs